Mage.Sets/src/mage/cards/p/PhialOfGaladriel.java

```diff
@@ -0,0 +1,140 @@
+
+        // If you would draw a card while you have no cards in hand, draw two cards instead.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new PhialOfGaladrielDrawEffect()));
+        // If you would gain life while you have 5 or less life, you gain twice that much life instead.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new PhialOfGaladrielLifeEffect()));
+
+        // {T}: Add one mana of any color.
+        this.addAbility(new AnyColorManaAbility());
+    }
+
+    private PhialOfGaladriel(final PhialOfGaladriel card) {
+        super(card);
+    }
+
+    @Override
+    public PhialOfGaladriel copy() {
+        return new PhialOfGaladriel(this);
+    }
+}
+
+class PhialOfGaladrielLifeEffect extends ReplacementEffectImpl {
+
+    PhialOfGaladrielLifeEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Benefit);
+        staticText = "If you would gain life while you have 5 or less life, " +
+            "you gain twice that much life instead";
+    }
+
+    private PhialOfGaladrielLifeEffect(final PhialOfGaladrielLifeEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PhialOfGaladrielLifeEffect copy() {
+        return new PhialOfGaladrielLifeEffect(this);
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        event.setAmount(CardUtil.overflowMultiply(event.getAmount(), 2));
+        return false;
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.GAIN_LIFE;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        UUID playerId = source.getControllerId();
+        if(playerId == null || !event.getPlayerId().equals(playerId)){
+            return false;
+        }
+
+        Player player = game.getPlayer(playerId);
+        if(player == null){
+            return false;
+        }
+
+        return player.getLife() <= 5;
+    }
+}
+
+class PhialOfGaladrielDrawEffect extends ReplacementEffectImpl {
+
+    PhialOfGaladrielDrawEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Neutral);
+        staticText = "If you would draw a card while you have no cards in hand, draw two cards instead";
+    }
+
+    private PhialOfGaladrielDrawEffect(final PhialOfGaladrielDrawEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PhialOfGaladrielDrawEffect copy() {
+        return new PhialOfGaladrielDrawEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null) {
+            controller.drawCards(2, source, game, event);
+        }
+        return true;
```

Comment:
It seems this is how other card draw replacement effects work, but I'm a bit suspicious of fully replacing events.

Can you test the interaction between this and e.g. Oneirophage to confirm?

Reply:
Will do.

Reply:
![oneirophage](https://github.com/magefree/mage/assets/34709007/f81454c2-6aee-4ca2-8126-5b7df5085ba3)

Seems to work fine to me. The draw 7 from Wheel of Fortune did get replaced as draw 2 + draw 6, and that did cause 8 Oneirophage triggers.

Reply:
Thanks for checking.

Reply:
@xenohedron it depends on event's implementations -- some replace events look to new values only, another for a possible "cancel". Only the last one depends on the replace's result.
![shot_230709_103356](https://github.com/magefree/mage/assets/8344157/81a32932-44d2-4bc0-a5f8-452e51df1c24)

So it's a good practice:
* for replace events: implement logic with a cancel style;
* for replace effects: return true on real replace (miss target, controller, permanent -- it's not a standard situation, so must return false, e.g. ignore replace and continue);
* for another effects in `apply` method: return true on real changes only (just a good code style for inheritance, you can still return true for hard tracking changes all the time);